libsolidity/analysis/TypeChecker.cpp

```diff
@@ -448,6 +448,9 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 	)
 		m_errorReporter.typeError(_variable.location(), "Variables cannot be declared in interfaces.");
 
+	if (_variable.typeName())
+		_variable.typeName()->accept(*this);
```

Comment:
Does this mean we're doing smtchecker traversal before the type checker? Or we do not resolve these in the type checker?